src/push.rs

```diff
@@ -80,6 +80,10 @@ fn push(job: &str,
         method: &str)
         -> Result<()> {
 
+    // Suppress clippy warning needless_pass_by_value.
+    let grouping = grouping;
+    let mfs = mfs;
```

Comment:
IMO, as a library, it should minimize its overhead, so when it does not need ownership, it should not require user to provide.